protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -677,26 +677,27 @@ class _FieldSet {
       return hash;
     }
 
-    int hashEachField(int hash) {
-      //non-extension fields
-      hash = _infosSortedByTag.where((fi) => _values[fi.index!] != null).fold(
-          hash, (int h, FieldInfo fi) => hashField(h, fi, _values[fi.index!]));
-
-      if (!_hasExtensions) return hash;
+    // Hash with descriptor.
+    var hash = _HashUtils._combine(0, _meta.hashCode);
 
-      hash =
-          _sorted(_extensions!._tagNumbers).fold(hash, (int h, int tagNumber) {
-        var fi = _extensions!._getInfoOrNull(tagNumber)!;
-        return hashField(h, fi, _extensions!._getFieldOrNull(fi));
-      });
+    // Hash with non-extension fields.
+    final values = _values;
+    for (final fi in _infosSortedByTag) {
```

Comment:
`_hashObjects`, called at line 671, also uses `fold`

Reply:
There is a big difference between these two patterns. The function passed to `fold()` in `_hashObjects` doesn't capture any local variables. I tested modifying that one as well and it was a regression in both JS and VM cases.